packages/buttons/src/styled/StyledExternalIcon.ts

```diff
@@ -7,17 +7,25 @@
 
 import styled from 'styled-components';
 import NewWindowIcon from '@zendeskgarden/svg-icons/src/12/new-window-stroke.svg';
-import { DEFAULT_THEME, retrieveComponentStyles } from '@zendeskgarden/react-theming';
+import { DEFAULT_THEME, retrieveComponentStyles, useText } from '@zendeskgarden/react-theming';
 
 const COMPONENT_ID = 'buttons.external_icon';
 
 /**
  * Accepts all `<svg>` props
  */
-export const StyledExternalIcon = styled(NewWindowIcon).attrs({
+export const StyledExternalIcon: any = styled(NewWindowIcon).attrs(props => ({
   'data-garden-id': COMPONENT_ID,
-  'data-garden-version': PACKAGE_VERSION
-})`
+  'data-garden-version': PACKAGE_VERSION,
+  role: 'img',
+  'aria-hidden': undefined,
+  'aria-label': useText(
+    StyledExternalIcon,
+    { 'aria-label': props['aria-label'] },
+    'aria-label',
+    '(opens in a new tab)'
+  )
+}))`
```

Comment:
Ideally, `useText` etc. should be applied at the element level within `Anchor.tsx`. The linked issue references a construct for handling conditionality within the hook call. I don't think we're expecting any changes to `StyledExternalIcon`.

Reply:
Hey @jzempel — when you say 
> I don't think we're expecting any changes to `StyledExternalIcon`
What do you mean? (Does the translated label not count as a change?)

Reply:
Could you please help me understand why? Reasoning about the code, I had been under the impression that the `<StyledExternalIcon>` component was the smallest/root piece (aside from the icon itself, from the icons package), which is why it made sense to me to make the changes there, rather than in the `Anchor.tsx` instance of it. 🤔

Reply:
As a rule, we want to see hooks applied at the [element API](https://github.com/zendeskgarden/react-components/blob/main/docs/api.md#element-components) level. The code here is unexpected, leading to unwanted maintenance challenges.

Reply:
Happy to make the requested change, to keep with codebase conventions. I guess what I'm wondering, at this point, is why the `<StyledExternalIcon>` component level doesn't qualify as the element API level?